skimage/measure/profile.py

```diff
@@ -109,3 +122,222 @@ def _line_profile_coordinates(src, dst, linewidth=1):
+        profile is the ceil of the computed length of the scan line.
+
+    Notes
+    -----
+    This is a utility method meant to be used internally by skimage functions.
+    The destination point is included in the profile, in contrast to
+    standard numpy indexing.
+    """
+    src_row, src_col, src_dim3 = src = np.asarray(src, dtype=float)
+    dst_row, dst_col, dst_dim3 = dst = np.asarray(dst, dtype=float)
+    d_row, d_col, d_plane = dst - src
+
+    # Get one unit vector perpendicular to direction vector to find a point
+    # that is one unit distance away from the destination vector
+    # (ex: ix + jy + kz = 0, then we can use x = y = 1)
+    # Try with z first if it is not 0, then the same for x, otherwise pick y
+    # We subtract 1 from linewidth to change from pixel-counting
+    # (make this line 3 pixels wide) to point distances (the
+    # distance between pixel centers)
+    if d_plane != 0:
+        # try finding the solution to ix + jy + kz = 0 for x = 1 and y = 1
+        dim_z = - (d_row + d_col) / d_plane
+        length_vector = np.sqrt(2 + dim_z ** 2)
+        col_width = row_width = (linewidth - 1) / (2 * length_vector)
+        slice_width = (linewidth - 1) * (dim_z / 2 * length_vector)
+    elif d_row != 0:
+        # try finding the solution to ix + jy + kz = 0 for y = 1 and z = 1
+        dim_x = - (d_plane + d_col) / d_row
+        length_vector = np.sqrt(2 + dim_x ** 2)
+        col_width = slice_width = (linewidth - 1) / (2 * length_vector)
+        row_width = (linewidth - 1) * (dim_x / length_vector) / 2
+    else:
+        # try finding the solution to ix + jy + kz = 0 for x = 1 and z = 1
+        dim_y = - (d_row + d_plane) / d_col
+        length_vector = np.sqrt(2 + dim_y ** 2)
+        row_width = slice_width = (linewidth - 1) / (2 * length_vector)
+        col_width = (linewidth - 1) * (dim_y / length_vector) / 2
+
+    # we add one above because we include the last point in the profile
+    # (in contrast to standard numpy indexing)
+    length = np.ceil(np.linalg.norm([d_row, d_col, d_plane]) + 1)
+
+    line_col = np.linspace(src_col, dst_col, length)
+    line_row = np.linspace(src_row, dst_row, length)
+    line_slice = np.linspace(src_dim3, dst_dim3, length)
+
+    perp_rows = np.array([np.linspace(row_i - row_width, row_i + row_width,
+                                      linewidth) for row_i in line_row])
+    perp_cols = np.array([np.linspace(col_i - col_width, col_i + col_width,
+                                      linewidth) for col_i in line_col])
+    perp_plane = np.array([np.linspace(slice_i - slice_width, slice_i + slice_width,
+                                       linewidth) for slice_i in line_slice])
+
+    perp_array = np.array([perp_rows, perp_cols, perp_plane])
+
+    # rotate all sample points around the direction axis if linewidth is > 1
+    if linewidth > 1:
+        perp_array = rotate_sample_points(linewidth, perp_array, src, dst)
```

Comment:
If I understand correctly, you're taking a sampling vector (e.g. of length 3) and then rotating it around the centerline, to get a larger sampling vector representing a disk. But, wouldn't this method cause a denser sampling near the centerline compared to the edge of the cylinder?